### PR TITLE
Don't show snippets in VB comments

### PR DESCRIPTION
**Pull request by chborl**
Fixes issue 21801

**Customer scenario**

Code snippets are displayed in VB comments, when snippet behavior is set to "Always include snippets".  

**Bugs this fixes:**

https://github.com/dotnet/roslyn/issues/21801

**Workarounds, if any**

Turn off code snippets for VB

**Risk**

Low risk bug fix that only affects the display of snippets in VB Comments

**Performance impact**

None, no complexity changes

**Is this a regression from a previous update?**
No

**Root cause analysis:**
Infrequently used option

**How was the bug found?**
Customer report

### CODE REVIEW

File: src/VisualStudio/Core/Test/Snippets/SnippetCompletionProviderTests.vb [hunk 1]
```diff
@@ -20,10 +18,10 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Snippets
             Using testState
                 testState.SendTabToCompletion()
 
-                Assert.Equal(testState.GetDocumentText(), "a")
+                Assert.Equal("a", testState.GetDocumentText())
```
Comment (rchande) [inline src/VisualStudio/Core/Test/Snippets/SnippetCompletionProviderTests.vb]: Nice catch!

File: src/VisualStudio/VisualBasic/Impl/Snippets/SnippetCompletionProvider.vb [hunk 2]
```diff
@@ -52,6 +54,11 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.Snippets
             Dim syntaxFacts = document.GetLanguageService(Of ISyntaxFactsService)()
             Dim isPossibleTupleContext = syntaxFacts.IsPossibleTupleContext(syntaxTree, position, cancellationToken)
 
+            Dim trivia = syntaxTree.FindTriviaToLeft(position, cancellationToken)
```
Comment (rchande) [inline src/VisualStudio/VisualBasic/Impl/Snippets/SnippetCompletionProvider.vb]: I think you want this helper: http://index/?leftProject=Microsoft.CodeAnalysis.VisualBasic.Workspaces&leftSymbol=oxebcswupkyz&file=Extensions%5CSyntaxTreeExtensions.vb

Comment (rchande) [reply src/VisualStudio/VisualBasic/Impl/Snippets/SnippetCompletionProvider.vb]: Actually, there's probably a similar method on the ISyntaxFactsService you have above.

**Comment by chborl** [comment]
Responded to feedback.  @dotnet/roslyn-ide

File: src/VisualStudio/VisualBasic/Impl/Snippets/SnippetCompletionProvider.vb [hunk 3]
```diff
@@ -52,6 +53,10 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.Snippets
             Dim syntaxFacts = document.GetLanguageService(Of ISyntaxFactsService)()
             Dim isPossibleTupleContext = syntaxFacts.IsPossibleTupleContext(syntaxTree, position, cancellationToken)
 
+            If (IsInNonUserCode(syntaxTree, position, cancellationToken)) Then
```
Comment (rchande) [inline src/VisualStudio/VisualBasic/Impl/Snippets/SnippetCompletionProvider.vb]: Nit: VB doesn't require parens around the expression if an if statement.

**Comment by dpoeschl** [comment]
Oops. Premature merging. Tagging @Pilchie @jinujoseph for Ask Mode.

**Comment by Pilchie** [comment]
Approved.

**Pull request merged**